code/modules/discord/discord_sql_functions.dm

```diff
@@ -0,0 +1,198 @@
+
+/**
+ * Generate a timebound token for discord verification
+ *
+ * This uses the common word list to generate a six word random token, this token can then be fed to a discord bot that has access
+ * to the same database, and it can use it to link a ckey to a discord id, with minimal user effort
+ *
+ * It returns the token to the calling proc, after inserting an entry into the discord_link table of the following form
+ *
+ * ```
+ * (unique_id, ckey, null, the current time, the one time token generated)
+ * the null value will be filled out with the discord id by the integrated discord bot when a user verifies
+ * ```
+ *
+ * Notes:
+ * * The token is guaranteed to be unique during it's validity period
+ * * The validity period is currently set at 4 hours
+ * * a token may not be unique outside it's validity window (to reduce conflicts)
+ *
+ * Arguments:
+ * * ckey_for a string representing the ckey this token is for
+ *
+ * Returns a string representing the one time token
+ */
+/client/proc/discord_generate_one_time_token(ckey_for)
+	var/static/list/token_wordlist = world.file2list("strings/eff_large_wordlist.txt")
```

Comment:
Why not just use `strings\1000_most_common.txt` rather than adding a new list?

Reply:
This was a direct cherry-pick. I could strip it out.

Reply:
Since the token is always guaranteed to be unique it won't matter that there aren't as many word options (unless someone gets really unlucky I guess), so using the list that's already there should work just as well.

Reply:
It's intended more as a security feature, I assume. As intent may be to reuse these in the future.